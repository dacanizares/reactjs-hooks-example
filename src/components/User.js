import React from 'react';
import Card from './Card';

const User = ({user}) =>  {   
    return (
        <div className='card-deck'>
            <Card text={user.email} title={`${user.name.title} ${user.name.last}`} imageUrl={user.picture.large} />    
        </div>
    );
}

export default User;