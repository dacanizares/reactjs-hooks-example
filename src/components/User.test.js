import React from 'react';
import User from './User';
import renderer from 'react-test-renderer';

jest.mock('./Card', () => 'Card');

it('renders correctly', () => {
  const user = {
    email: 'email@test.com', 
    name: {
      title: 'mr',
      first: 'name'
    },
    picture: {
      large: 'image here'
    }
  }

  const tree = renderer
    .create(<User user={user} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});