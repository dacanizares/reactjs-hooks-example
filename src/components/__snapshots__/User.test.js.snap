// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card-deck"
>
  <Card
    imageUrl="image here"
    text="email@test.com"
    title="mr undefined"
  />
</div>
`;
